src/neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -112,6 +139,24 @@ protected override void OnPersist(ApplicationEngine engine)
             var account = Contract.CreateSignatureRedeemScript(pubkey).ToScriptHash();
             GAS.Mint(engine, account, gasPerBlock * CommitteeRewardRatio / 100);
 
+            // Record the cumulative reward of the voters of each committee
+
+            var voterRewardPerCommittee = gasPerBlock * VoterRewardRatio / 100 / (ProtocolSettings.Default.CommitteeMembersCount + ProtocolSettings.Default.ValidatorsCount);
+            var committeeVotes = GetCommitteeVotes(engine.Snapshot).ToArray();
+            for (var i = 0; i < committeeVotes.Length; i++)
+            {
+                if (committeeVotes[i].Votes > 0)
+                {
+                    BigInteger voterSumRewardPerNEO = (i < ProtocolSettings.Default.ValidatorsCount ? 2 : 1) * voterRewardPerCommittee * 10000L / committeeVotes[i].Votes; // Zoom in 10000 times, and the final calculation should be divided 10000L
+                    var voterRewardKeyPrefix = CreateStorageKey(Prefix_VoterRewardPerCommittee).Add(committeeVotes[i].PublicKey);
+                    var enumerator = engine.Snapshot.Storages.Find(voterRewardKeyPrefix.ToArray()).GetEnumerator();
+                    if (enumerator.MoveNext())
+                        voterSumRewardPerNEO += new BigInteger(enumerator.Current.Value.Value);
+                    var voterRewardKey = CreateStorageKey(Prefix_VoterRewardPerCommittee).Add(committeeVotes[i].PublicKey).AddBigEndian(uint.MaxValue - engine.Snapshot.PersistingBlock.Index - 1);
+                    engine.Snapshot.Storages.Add(voterRewardKey, new StorageItem() { Value = voterSumRewardPerNEO.ToByteArray() });
+                }
+            }
```

Comment:
21 new records per block?

Reply:
Yes, we need to keep track of the cumulative gains of each committee.

Reply:
I am worried that there will be too much data.

Reply:
One thing we could probably do is to take an average of votes every 100 blocks, something like that. It would be a good approximation for GAS distributions. 
Let's thing about other possibilities as well.

Reply:
Another way is only to record it when the reward value changes. I'll have a try.

Reply:
It will change every block. Because the value is accumulated.

Reply:
Not only that, but also lots of new reads. I'm wondering if anyone tested this changeset for performance implications because just looking at the number or storage reads/writes the impact must be quite noticeable, especially for block imports.

Reply:
the disk consumption is 294M for CN node and 200+M for normal node, (Data_LevelDB_ xxx folder), when generate 77000+ blocks, the preview3 version consumes only 30M for 110000+ blocks.

Reply:
Maybe we can spread the records into 21 blocks instead. In each block we keep the record of the (Height % 28 % 21)th committee member, and grant the incentive of voters of this specified committee. On the other hand, we calculate votes every 28 blocks to determine new committee members. By doing so not only numbers of records are less but also we can decrease calculation on voting.

Reply:
For example, we grant the incentive of voters of the 4th committee members in Height 4 as 4 % 28 % 21 = 4. In Height 25 we will grant the incentive of voters of the 4th committee members again as 25 % 28 % 21 = 4. The reason we grant incentive of voters of the 4th committee member is because the 4th committee member is also consensus node. In Height 28 we recalculate committee members. In Height 32 we will grant the incentive of the 4th committee member because 32 % 28 % 21 = 4, but this 4th committee member is a different one.

Reply:
> Maybe we can spread the records into 21 blocks instead. In each block we keep the record of the (Height % 28 % 21)th committee member, and grant the incentive of voters of this specified committee. On the other hand, we calculate votes every 28 blocks to determine new committee members. By doing so not only numbers of records are less but also we can decrease calculation on voting.

It's a good idea. What do you think? @neo-project/core @neo-project/ngd-shanghai

Reply:
Looks good.

Reply:
Done, but it may have the same problem with https://github.com/neo-project/neo/pull/1907#discussion_r484284238